tools/federation/src/main/java/org/eclipse/rdf4j/federated/write/WriteStrategy.java

```diff
@@ -25,49 +24,35 @@
  * @see RepositoryWriteStrategy
  * @see ReadOnlyWriteStrategy
  */
-public interface WriteStrategy {
-
-	/**
-	 * Initialize the write strategy (e.g. open a shared {@link RepositoryConnection}.
-	 * 
-	 * @throws RepositoryException
-	 */
-	public void initialize() throws RepositoryException;
-
-	/**
-	 * Returns true if this instance is initialized
-	 * 
-	 * @return flag indicating the initialization resources
-	 */
-	public boolean isInitialized();
-
+public interface WriteStrategy extends AutoCloseable {
```

Comment:
we should mention here that an instance should be specifically bound to a RepositoryConnection, at least as internal comment